java/statestore/src/test/java/sleeper/statestore/transactionlog/TransactionLogSnapshotTestBase.java

```diff
@@ -0,0 +1,221 @@
+import org.testcontainers.utility.DockerImageName;
+
+import sleeper.configuration.properties.instance.InstanceProperties;
+import sleeper.configuration.properties.table.TableProperties;
+import sleeper.configuration.properties.table.TableProperty;
+import sleeper.core.CommonTestConstants;
+import sleeper.core.schema.Schema;
+import sleeper.core.schema.type.LongType;
+import sleeper.core.statestore.StateStore;
+import sleeper.core.statestore.transactionlog.InMemoryTransactionLogStore;
+import sleeper.core.statestore.transactionlog.TransactionLogStateStore;
+import sleeper.core.statestore.transactionlog.TransactionLogStore;
+import sleeper.io.parquet.utils.HadoopConfigurationLocalStackUtils;
+import sleeper.statestore.transactionlog.DynamoDBTransactionLogSnapshotStore.LatestSnapshotsMetadataLoader;
+import sleeper.statestore.transactionlog.DynamoDBTransactionLogSnapshotStore.SnapshotMetadataSaver;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static sleeper.configuration.properties.InstancePropertiesTestHelper.createTestInstanceProperties;
+import static sleeper.configuration.properties.instance.CdkDefinedInstanceProperty.DATA_BUCKET;
+import static sleeper.configuration.properties.instance.CommonProperty.FILE_SYSTEM;
+import static sleeper.configuration.properties.table.TablePropertiesTestHelper.createTestTableProperties;
+import static sleeper.configuration.properties.table.TableProperty.STATESTORE_CLASSNAME;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_ID;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
+import static sleeper.configuration.testutils.LocalStackAwsV1ClientHelper.buildAwsV1Client;
+import static sleeper.core.schema.SchemaTestHelper.schemaWithKey;
+import static sleeper.core.statestore.FileReferenceTestData.DEFAULT_UPDATE_TIME;
+
+@Testcontainers
+public class TransactionLogSnapshotTestBase {
+    @TempDir
+    private java.nio.file.Path tempDir;
+    private FileSystem fs;
+    protected final Schema schema = schemaWithKey("key", new LongType());
+    private final Map<String, TransactionLogStore> partitionTransactionStoreByTableId = new HashMap<>();
+    private final Map<String, TransactionLogStore> fileTransactionStoreByTableId = new HashMap<>();
+    @Container
+    public static LocalStackContainer localStackContainer = new LocalStackContainer(DockerImageName.parse(CommonTestConstants.LOCALSTACK_DOCKER_IMAGE))
+            .withServices(LocalStackContainer.Service.S3, LocalStackContainer.Service.DYNAMODB);
+    protected static AmazonDynamoDB dynamoDBClient;
+    protected static AmazonS3 s3Client;
+    protected final InstanceProperties instanceProperties = createTestInstanceProperties();
+    protected final Configuration configuration = HadoopConfigurationLocalStackUtils.getHadoopConfiguration(localStackContainer);
```

Comment:
This is pointing it to LocalStack but then below it's set to use the local file system. Do we need S3 enabled in LocalStack?

Reply:
Oh, I see we're using the DynamoDB implementation of the state store for some tests.